src/procstat.c

```diff
@@ -886,7 +877,16 @@ static int register_item(struct procstat_context *context,
 	item->parent = parent;
 
 	if (parent) {
-		procstat_unlock(parent);
+		if (item_type_directory(item) && root_directory(context, parent)) {
+			// Each directory under the root has its own mutex
+			item->mutexptr = procstat_retain_mutex_from_pool();
```

Comment:
As we only produce new locks here while modifying the root and holding the root lock, the atomic implementation of procstat_retain_mutex_from_pool() looks redundant.